docs/3.0.0-beta.0/guides/authentication.md

```diff
@@ -0,0 +1,346 @@
+- [Github](https://github.com/strapi/strapi-examples/blob/master/login-react/doc/github_setup.md)
+- [Twitter](https://github.com/strapi/strapi-examples/blob/master/login-react/doc/twitter_setup.md)
+
+[👀   See our complete example with detailed tutorials for each provider (with React)](https://github.com/strapi/strapi-examples/tree/master/login-react)
+
+---
+
+Set your providers credentials in the admin interface (Plugin Users & Permissions > Providers).
+Then update and enable the provider you want use.
+
+To authenticate the user, use the GET method to request the url, `/connect/:provider`. eg: `GET /connect/facebook`
+
+After authentication, create and customize your own redirect callback at `/auth/:provider/callback`. The `jwt` and `user` data will be available in a .json response.
+
+Response payload:
+
+```json
+{
+  "user": {},
+  "jwt": ""
+}
+```
+
+## Forgotten password
+
+This action sends an email to a user with the link of you reset password page. This link contains an URL param `code` which is required to reset user password.
+
+#### Usage
+
+- `email` is your user email.
+- `url` is the url link that user will receive. After the user triggers a new password reset, 
+it is used to redirect the user to the new-password form. 
+
+```js
+import axios from 'axios';
+
+// Request API.
+axios
+  .post('http://localhost:1337/auth/forgot-password', {
+    email: 'user@strapi.io',
+    url: 'http:/localhost:1337/admin/plugins/users-permissions/auth/reset-password'
+  })
+  .then(response => {
+    // Handle success.
+    console.log('Your user received an email');
+  })
+  .catch(error => {
+    // Handle error.
+    console.log('An error occurred:', error);
+  });
+```
+
+## Password reset
+
+This action will reset the user password.
+
+#### Usage
+
+- `code` is the url params received from the email link (see forgot password)
+
+```js
+import axios from 'axios';
+
+// Request API.
+axios
+  .post('http://localhost:1337/auth/reset-password', {
+    code: 'privateCode',
+    password: 'myNewPassword',
+    passwordConfirmation: 'myNewPassword'
+  })
+  .then(response => {
+    // Handle success.
+    console.log('Your user\'s password has been changed.');
+  })
+  .catch(error => {
+    // Handle error.
+    console.log('An error occurred:', error);
+  });
+});
+```
+
+## User object in Strapi context
+
+The `user` object is available to successfully authenticated requests.
+
+#### Usage
+- The authenticated `user` object is a property of `ctx.state`.
+
+
+```js
+  create: async (ctx) => {
+
+    const { _id } = ctx.state.user
+
+    const depositObj = {
+      ...ctx.request.body,
+      depositor: _id
+    }
+
+    const data = await strapi.services.deposit.add(depositObj);
+
+    // Send 201 `created`
+    ctx.created(data);
+  }
+
+```
+
+## Adding a new provider
+
+To add a new provider on Strapi, you will need to perform changes onto the following files:
+
+```
+packages/strapi-plugin-users-permissions/services/Providers.js
```

Comment:
Paths need to be changed for beta rebuilding steps, assuming since no more providers are accepted via PR